x-pack/legacy/plugins/ml/public/contexts/kibana/use_kibana_context.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useContext } from 'react';
+
+import { KibanaContext, KibanaContextValue } from './kibana_context';
+
+export const useKibanaContext = () => {
+  const context = useContext(KibanaContext);
+
+  if (
+    context.combinedQuery === undefined ||
+    context.currentIndexPattern === undefined ||
+    context.currentSavedSearch === undefined ||
+    context.indexPatterns === undefined ||
+    context.kbnBaseUrl === undefined ||
+    context.kibanaConfig === undefined
+  ) {
+    throw new Error('required attribute is undefined');
```

Comment:
Might be useful here to loop through the required attributes to check for undefined and then throw the error with the specific missing dependency. Just for future debugging purposes.

Reply:
this would potentially throw multiple times if more than one property is missing unless the loop builds up an array of missing properties and joins them for the error.

Reply:
Yes, the second thing you said. Should've been more clear. Not a must-have but would be nice to have.